### PR TITLE
Ramlog: recover last crash log from ram buffer.

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
Ramlog: recover last crash log from ram buffer.
We need place ram buffer to a fixed section, it will not be initialized on boot.
## Testing
daily test.

### CODE REVIEW

File: drivers/syslog/Kconfig [hunk 1]
```diff
@@ -34,6 +34,14 @@ config RAMLOG
 		details as needed to support logging.
 
 if RAMLOG
+config RAMLOG_BUFFER_SECTION
+	string "The section where ramlog buffer is located"
+	default ".bss"
+	depends on RAMLOG_SYSLOG
+	---help---
+		The section where ramlog buffer is located, this section cannot
+		be initialized each time when the system boot.
```
Comment (gustavonihei) [inline drivers/syslog/Kconfig:43]: nit: Suggestion for trying to improve the help text.
```suggestion
		The section where ramlog buffer is located.
		This section shall not be initialized on system boot.
```

File: drivers/syslog/ramlog.c [hunk 2]
```diff
@@ -639,6 +643,60 @@ int ramlog_poll(FAR struct file *filep, FAR struct pollfd *fds, bool setup)
   return ret;
 }
 
+/****************************************************************************
+ * Name: ramlog_initbuf
+ *
+ * Description:
+ *  Initialize g_sysdev based on the current system ramlog buffer.
```
Comment (gustavonihei) [inline drivers/syslog/ramlog.c:650]:
```suggestion
 *   Initialize g_sysdev based on the current system ramlog buffer.
```

File: drivers/syslog/ramlog.c [hunk 3]
```diff
@@ -639,6 +643,60 @@ int ramlog_poll(FAR struct file *filep, FAR struct pollfd *fds, bool setup)
...
+          priv->rl_tail = i;
+        }
+
+      prev = cur;
+    }
+
+out:
+  if (i != priv->rl_bufsize)
+    {
+      priv->rl_head = priv->rl_tail = 0;
```
Comment (gustavonihei) [inline drivers/syslog/ramlog.c:694]:
```suggestion
      priv->rl_head = 0;
      priv->rl_tail = 0;
```
This is a violation to the One Statement per Line rule of the coding standard:
https://cwiki.apache.org/confluence/display/NUTTX/Coding+Standard#onestatement

Comment (Donny9) [reply drivers/syslog/ramlog.c]: Done. Thank you.

File: drivers/syslog/ramlog.c (hunk 2 above)
Comment (Donny9) [reply drivers/syslog/ramlog.c]: Done. Thank you.

File: drivers/syslog/Kconfig (hunk 1 above)
Comment (Donny9) [reply drivers/syslog/Kconfig]: Done. Thank you.

**Comment by xiaoxiang781216** [comment]
@gustavonihei Do you have more comment?

**Review by gustavonihei — APPROVED** [review]
LGTM

**Pull request merged**